Tests failures using 2018.1 CE

Note that this is CE only (no issues with UE).

```
com.google.cloud.tools.intellij.startup.CloudToolsPluginInitializationComponentTest > testInitComponent_feedbackIsDisabled FAILED
    java.lang.AssertionError: Cannot load resource: /idea/ApplicationInfo.xml
        at com.intellij.openapi.diagnostic.DefaultLogger.error(DefaultLogger.java:77)
        at com.intellij.openapi.diagnostic.Logger.error(Logger.java:153)
        at com.intellij.ide.plugins.PluginManagerCore.initPlugins(PluginManagerCore.java:1492)
        at com.intellij.ide.plugins.PluginManagerCore.getPlugins(PluginManagerCore.java:130)
        at com.intellij.ide.plugins.PluginManagerCore.getPlugins(PluginManagerCore.java:124)
        at com.intellij.ide.plugins.PluginManager.getPlugin(PluginManager.java:221)
        at com.google.cloud.tools.intellij.startup.GoogleAccountPluginUninstaller.uninstallIfPresent(GoogleAccountPluginUninstaller.java:47)
        at com.google.cloud.tools.intellij.startup.CloudToolsPluginInitializationComponent.initComponent(CloudToolsPluginInitializationComponent.java:65)
        at com.google.cloud.tools.intellij.startup.CloudToolsPluginInitializationComponentTest.testInitComponent_feedbackIsDisabled(CloudToolsPluginInitializationComponentTest.java:77)

        Caused by:
        java.lang.RuntimeException: Cannot load resource: /idea/ApplicationInfo.xml
            at com.intellij.openapi.application.impl.ApplicationInfoImpl.<init>(ApplicationInfoImpl.java:216)
            at com.intellij.openapi.application.impl.ApplicationInfoImpl.getShadowInstance(ApplicationInfoImpl.java:625)
            at com.intellij.ide.plugins.PluginManagerCore.checkEssentialPluginsAreAvailable(PluginManagerCore.java:1299)
            at com.intellij.ide.plugins.PluginManagerCore.initializePlugins(PluginManagerCore.java:1311)
            at com.intellij.ide.plugins.PluginManagerCore.initPlugins(PluginManagerCore.java:1486)
            ... 6 more

            Caused by:
            java.io.FileNotFoundException: /idea/ApplicationInfo.xml
                at com.intellij.openapi.util.JDOMUtil.loadDocument(JDOMUtil.java:311)
                at com.intellij.openapi.application.impl.ApplicationInfoImpl.<init>(ApplicationInfoImpl.java:212)
                ... 10 more

com.google.cloud.tools.intellij.startup.CloudToolsPluginInitializationComponentTest > testInitComponent_loginServiceIsInitialized FAILED
    java.lang.AssertionError: Cannot load resource: /idea/ApplicationInfo.xml
        at com.intellij.openapi.diagnostic.DefaultLogger.error(DefaultLogger.java:77)
        at com.intellij.openapi.diagnostic.Logger.error(Logger.java:153)
        at com.intellij.ide.plugins.PluginManagerCore.initPlugins(PluginManagerCore.java:1492)
        at com.intellij.ide.plugins.PluginManagerCore.getPlugins(PluginManagerCore.java:130)
        at com.intellij.ide.plugins.PluginManagerCore.getPlugins(PluginManagerCore.java:124)
        at com.intellij.ide.plugins.PluginManager.getPlugin(PluginManager.java:221)
        at com.google.cloud.tools.intellij.startup.GoogleAccountPluginUninstaller.uninstallIfPresent(GoogleAccountPluginUninstaller.java:47)
        at com.google.cloud.tools.intellij.startup.CloudToolsPluginInitializationComponent.initComponent(CloudToolsPluginInitializationComponent.java:65)
        at com.google.cloud.tools.intellij.startup.CloudToolsPluginInitializationComponentTest.testInitComponent_loginServiceIsInitialized(CloudToolsPluginInitializationComponentTest.java:104)

        Caused by:
        java.lang.RuntimeException: Cannot load resource: /idea/ApplicationInfo.xml
            at com.intellij.openapi.application.impl.ApplicationInfoImpl.<init>(ApplicationInfoImpl.java:216)
            at com.intellij.openapi.application.impl.ApplicationInfoImpl.getShadowInstance(ApplicationInfoImpl.java:625)
            at com.intellij.ide.plugins.PluginManagerCore.checkEssentialPluginsAreAvailable(PluginManagerCore.java:1299)
            at com.intellij.ide.plugins.PluginManagerCore.initializePlugins(PluginManagerCore.java:1311)
            at com.intellij.ide.plugins.PluginManagerCore.initPlugins(PluginManagerCore.java:1486)
            ... 6 more

            Caused by:
            java.io.FileNotFoundException: /idea/ApplicationInfo.xml
                at com.intellij.openapi.util.JDOMUtil.loadDocument(JDOMUtil.java:311)
                at com.intellij.openapi.application.impl.ApplicationInfoImpl.<init>(ApplicationInfoImpl.java:212)
                ... 10 more

com.google.cloud.tools.intellij.startup.CloudToolsPluginInitializationComponentTest > testInitComponent_usageTrackingIsEnabled FAILED
    java.lang.AssertionError: Cannot load resource: /idea/ApplicationInfo.xml
        at com.intellij.openapi.diagnostic.DefaultLogger.error(DefaultLogger.java:77)
        at com.intellij.openapi.diagnostic.Logger.error(Logger.java:153)
        at com.intellij.ide.plugins.PluginManagerCore.initPlugins(PluginManagerCore.java:1492)
        at com.intellij.ide.plugins.PluginManagerCore.getPlugins(PluginManagerCore.java:130)
        at com.intellij.ide.plugins.PluginManagerCore.getPlugins(PluginManagerCore.java:124)
        at com.intellij.ide.plugins.PluginManager.getPlugin(PluginManager.java:221)
        at com.google.cloud.tools.intellij.startup.GoogleAccountPluginUninstaller.uninstallIfPresent(GoogleAccountPluginUninstaller.java:47)
        at com.google.cloud.tools.intellij.startup.CloudToolsPluginInitializationComponent.initComponent(CloudToolsPluginInitializationComponent.java:65)
        at com.google.cloud.tools.intellij.startup.CloudToolsPluginInitializationComponentTest.testInitComponent_usageTrackingIsEnabled(CloudToolsPluginInitializationComponentTest.java:88)

        Caused by:
        java.lang.RuntimeException: Cannot load resource: /idea/ApplicationInfo.xml
            at com.intellij.openapi.application.impl.ApplicationInfoImpl.<init>(ApplicationInfoImpl.java:216)
            at com.intellij.openapi.application.impl.ApplicationInfoImpl.getShadowInstance(ApplicationInfoImpl.java:625)
            at com.intellij.ide.plugins.PluginManagerCore.checkEssentialPluginsAreAvailable(PluginManagerCore.java:1299)
            at com.intellij.ide.plugins.PluginManagerCore.initializePlugins(PluginManagerCore.java:1311)
            at com.intellij.ide.plugins.PluginManagerCore.initPlugins(PluginManagerCore.java:1486)
            at com.intellij.ide.plugins.PluginManagerCore.getPlugins(PluginManagerCore.java:130)
            at com.intellij.ide.plugins.PluginManagerCore.getPlugins(PluginManagerCore.java:124)
            at com.intellij.ide.plugins.PluginManager.getPlugin(PluginManager.java:221)
            at com.google.cloud.tools.intellij.startup.GoogleAccountPluginUninstaller.uninstallIfPresent(GoogleAccountPluginUninstaller.java:47)
            at com.google.cloud.tools.intellij.startup.CloudToolsPluginInitializationComponent.initComponent(CloudToolsPluginInitializationComponent.java:65)
            at com.google.cloud.tools.intellij.startup.CloudToolsPluginInitializationComponent$MockitoMock$214027090.initComponent$accessor$3HKpaixT(Unknown Source)
            at com.google.cloud.tools.intellij.startup.CloudToolsPluginInitializationComponent$MockitoMock$214027090$auxiliary$9jIrPCpk.call(Unknown Source)
            at org.mockito.internal.invocation.RealMethod$FromCallable$1.call(RealMethod.java:40)
            at org.mockito.internal.invocation.RealMethod$FromBehavior.invoke(RealMethod.java:62)
            at org.mockito.internal.invocation.InterceptedInvocation.callRealMethod(InterceptedInvocation.java:127)
            at org.mockito.internal.stubbing.answers.CallsRealMethods.answer(CallsRealMethods.java:43)
            at org.mockito.Answers.answer(Answers.java:100)
            at org.mockito.internal.handler.MockHandlerImpl.handle(MockHandlerImpl.java:104)
            at org.mockito.internal.handler.NullResultGuardian.handle(NullResultGuardian.java:29)
            at org.mockito.internal.handler.InvocationNotifierHandler.handle(InvocationNotifierHandler.java:35)
            at org.mockito.internal.creation.bytebuddy.MockMethodInterceptor.doIntercept(MockMethodInterceptor.java:63)
            at org.mockito.internal.creation.bytebuddy.MockMethodInterceptor.doIntercept(MockMethodInterceptor.java:49)
            at org.mockito.internal.creation.bytebuddy.MockMethodInterceptor$DispatcherDefaultingToRealMethod.interceptSuperCallable(MockMethodInterceptor.java:110)
            at com.google.cloud.tools.intellij.startup.CloudToolsPluginInitializationComponent$MockitoMock$214027090.initComponent(Unknown Source)
            ... 1 more

            Caused by:
            java.io.FileNotFoundException: /idea/ApplicationInfo.xml
                at com.intellij.openapi.util.JDOMUtil.loadDocument(JDOMUtil.java:311)
                at com.intellij.openapi.application.impl.ApplicationInfoImpl.<init>(ApplicationInfoImpl.java:212)
                ... 24 more

com.google.cloud.tools.intellij.startup.CloudToolsPluginInitializationComponentTest > testInitComponent_feedbackIsEnabled FAILED
    java.lang.AssertionError: Cannot load resource: /idea/ApplicationInfo.xml
        at com.intellij.openapi.diagnostic.DefaultLogger.error(DefaultLogger.java:77)
        at com.intellij.openapi.diagnostic.Logger.error(Logger.java:153)
        at com.intellij.ide.plugins.PluginManagerCore.initPlugins(PluginManagerCore.java:1492)
        at com.intellij.ide.plugins.PluginManagerCore.getPlugins(PluginManagerCore.java:130)
        at com.intellij.ide.plugins.PluginManagerCore.getPlugins(PluginManagerCore.java:124)
        at com.intellij.ide.plugins.PluginManager.getPlugin(PluginManager.java:221)
        at com.google.cloud.tools.intellij.startup.GoogleAccountPluginUninstaller.uninstallIfPresent(GoogleAccountPluginUninstaller.java:47)
        at com.google.cloud.tools.intellij.startup.CloudToolsPluginInitializationComponent.initComponent(CloudToolsPluginInitializationComponent.java:65)
        at com.google.cloud.tools.intellij.startup.CloudToolsPluginInitializationComponentTest.testInitComponent_feedbackIsEnabled(CloudToolsPluginInitializationComponentTest.java:70)

        Caused by:
        java.lang.RuntimeException: Cannot load resource: /idea/ApplicationInfo.xml
            at com.intellij.openapi.application.impl.ApplicationInfoImpl.<init>(ApplicationInfoImpl.java:216)
            at com.intellij.openapi.application.impl.ApplicationInfoImpl.getShadowInstance(ApplicationInfoImpl.java:625)
            at com.intellij.ide.plugins.PluginManagerCore.checkEssentialPluginsAreAvailable(PluginManagerCore.java:1299)
            at com.intellij.ide.plugins.PluginManagerCore.initializePlugins(PluginManagerCore.java:1311)
            at com.intellij.ide.plugins.PluginManagerCore.initPlugins(PluginManagerCore.java:1486)
            ... 6 more

            Caused by:
            java.io.FileNotFoundException: /idea/ApplicationInfo.xml
                at com.intellij.openapi.util.JDOMUtil.loadDocument(JDOMUtil.java:311)
                at com.intellij.openapi.application.impl.ApplicationInfoImpl.<init>(ApplicationInfoImpl.java:212)
                ... 10 more

com.google.cloud.tools.intellij.startup.CloudToolsPluginInitializationComponentTest > testInitComponent_usageTrackingIsDisabled FAILED
    java.lang.AssertionError: Cannot load resource: /idea/ApplicationInfo.xml
        at com.intellij.openapi.diagnostic.DefaultLogger.error(DefaultLogger.java:77)
        at com.intellij.openapi.diagnostic.Logger.error(Logger.java:153)
        at com.intellij.ide.plugins.PluginManagerCore.initPlugins(PluginManagerCore.java:1492)
        at com.intellij.ide.plugins.PluginManagerCore.getPlugins(PluginManagerCore.java:130)
        at com.intellij.ide.plugins.PluginManagerCore.getPlugins(PluginManagerCore.java:124)
        at com.intellij.ide.plugins.PluginManager.getPlugin(PluginManager.java:221)
        at com.google.cloud.tools.intellij.startup.GoogleAccountPluginUninstaller.uninstallIfPresent(GoogleAccountPluginUninstaller.java:47)
        at com.google.cloud.tools.intellij.startup.CloudToolsPluginInitializationComponent.initComponent(CloudToolsPluginInitializationComponent.java:65)
        at com.google.cloud.tools.intellij.startup.CloudToolsPluginInitializationComponentTest.testInitComponent_usageTrackingIsDisabled(CloudToolsPluginInitializationComponentTest.java:98)

        Caused by:
        java.lang.RuntimeException: Cannot load resource: /idea/ApplicationInfo.xml
            at com.intellij.openapi.application.impl.ApplicationInfoImpl.<init>(ApplicationInfoImpl.java:216)
            at com.intellij.openapi.application.impl.ApplicationInfoImpl.getShadowInstance(ApplicationInfoImpl.java:625)
            at com.intellij.ide.plugins.PluginManagerCore.checkEssentialPluginsAreAvailable(PluginManagerCore.java:1299)
            at com.intellij.ide.plugins.PluginManagerCore.initializePlugins(PluginManagerCore.java:1311)
            at com.intellij.ide.plugins.PluginManagerCore.initPlugins(PluginManagerCore.java:1486)
            at com.intellij.ide.plugins.PluginManagerCore.getPlugins(PluginManagerCore.java:130)
            at com.intellij.ide.plugins.PluginManagerCore.getPlugins(PluginManagerCore.java:124)
            at com.intellij.ide.plugins.PluginManager.getPlugin(PluginManager.java:221)
            at com.google.cloud.tools.intellij.startup.GoogleAccountPluginUninstaller.uninstallIfPresent(GoogleAccountPluginUninstaller.java:47)
            at com.google.cloud.tools.intellij.startup.CloudToolsPluginInitializationComponent.initComponent(CloudToolsPluginInitializationComponent.java:65)
            at com.google.cloud.tools.intellij.startup.CloudToolsPluginInitializationComponent$MockitoMock$214027090.initComponent$accessor$3HKpaixT(Unknown Source)
            at com.google.cloud.tools.intellij.startup.CloudToolsPluginInitializationComponent$MockitoMock$214027090$auxiliary$9jIrPCpk.call(Unknown Source)
            at org.mockito.internal.invocation.RealMethod$FromCallable$1.call(RealMethod.java:40)
            at org.mockito.internal.invocation.RealMethod$FromBehavior.invoke(RealMethod.java:62)
            at org.mockito.internal.invocation.InterceptedInvocation.callRealMethod(InterceptedInvocation.java:127)
            at org.mockito.internal.stubbing.answers.CallsRealMethods.answer(CallsRealMethods.java:43)
            at org.mockito.Answers.answer(Answers.java:100)
            at org.mockito.internal.handler.MockHandlerImpl.handle(MockHandlerImpl.java:104)
            at org.mockito.internal.handler.NullResultGuardian.handle(NullResultGuardian.java:29)
            at org.mockito.internal.handler.InvocationNotifierHandler.handle(InvocationNotifierHandler.java:35)
            at org.mockito.internal.creation.bytebuddy.MockMethodInterceptor.doIntercept(MockMethodInterceptor.java:63)
            at org.mockito.internal.creation.bytebuddy.MockMethodInterceptor.doIntercept(MockMethodInterceptor.java:49)
            at org.mockito.internal.creation.bytebuddy.MockMethodInterceptor$DispatcherDefaultingToRealMethod.interceptSuperCallable(MockMethodInterceptor.java:110)
            at com.google.cloud.tools.intellij.startup.CloudToolsPluginInitializationComponent$MockitoMock$214027090.initComponent(Unknown Source)
            ... 1 more

            Caused by:
            java.io.FileNotFoundException: /idea/ApplicationInfo.xml
                at com.intellij.openapi.util.JDOMUtil.loadDocument(JDOMUtil.java:311)
                at com.intellij.openapi.application.impl.ApplicationInfoImpl.<init>(ApplicationInfoImpl.java:212)

```

--- a/google-cloud-tools-plugin/testSrc/com/google/cloud/tools/intellij/startup/CloudToolsPluginInitializationComponentTest.java
+++ b/google-cloud-tools-plugin/testSrc/com/google/cloud/tools/intellij/startup/CloudToolsPluginInitializationComponentTest.java
@@ -16,9 +16,8 @@
 
 package com.google.cloud.tools.intellij.startup;
 
-import static org.mockito.Matchers.anyString;
+import static org.mockito.ArgumentMatchers.anyString;
 import static org.mockito.Mockito.doNothing;
-import static org.mockito.Mockito.mock;
 import static org.mockito.Mockito.never;
 import static org.mockito.Mockito.spy;
 import static org.mockito.Mockito.verify;
@@ -28,39 +27,41 @@ import com.google.cloud.tools.intellij.login.IntegratedGoogleLoginService;
 import com.google.cloud.tools.intellij.service.ApplicationPluginInfoService;
 import com.google.cloud.tools.intellij.service.PluginConfigurationService;
 import com.google.cloud.tools.intellij.service.PluginInfoService;
-import com.google.cloud.tools.intellij.testing.BasePluginTestCase;
+import com.google.cloud.tools.intellij.testing.CloudToolsRule;
+import com.google.cloud.tools.intellij.testing.TestService;
 import com.intellij.openapi.Disposable;
-import com.intellij.openapi.actionSystem.ActionManager;
 import com.intellij.openapi.application.Application;
 import com.intellij.openapi.application.ApplicationManager;
+import org.junit.After;
 import org.junit.Before;
+import org.junit.Rule;
 import org.junit.Test;
-import org.junit.runner.RunWith;
 import org.mockito.Mock;
-import org.mockito.junit.MockitoJUnitRunner;
 
 /** Tests to validate initialization on supported platforms */
-@RunWith(MockitoJUnitRunner.class)
-public class CloudToolsPluginInitializationComponentTest extends BasePluginTestCase {
+public class CloudToolsPluginInitializationComponentTest {
+
+  @Rule public final CloudToolsRule cloudToolsRule = new CloudToolsRule(this);
 
   private static final String PLUGIN_ID_STRING = "com.google.gct.core";
-  @Mock PluginInfoService pluginInfoService;
-  @Mock PluginConfigurationService pluginConfigurationService;
-  @Mock ActionManager actionManager;
-  @Mock ApplicationPluginInfoService applicationInfoService;
-  @Mock IntegratedGoogleLoginService googleLoginService;
+  @Mock Disposable disposable;
+  @Mock @TestService PluginInfoService pluginInfoService;
+  @Mock @TestService PluginConfigurationService pluginConfigurationService;
+  @Mock @TestService ApplicationPluginInfoService applicationInfoService;
+  @Mock @TestService IntegratedGoogleLoginService googleLoginService;
 
+  private Application application;
   CloudToolsPluginInitializationComponent testComponent;
 
   @Before
   public void registerMockServices() {
-    registerService(PluginInfoService.class, pluginInfoService);
-    registerService(PluginConfigurationService.class, pluginConfigurationService);
-    registerService(ActionManager.class, actionManager);
-    registerService(ApplicationPluginInfoService.class, applicationInfoService);
-    registerService(IntegratedGoogleLoginService.class, googleLoginService);
-
     testComponent = new CloudToolsPluginInitializationComponent();
+    application = ApplicationManager.getApplication();
+  }
+
+  @After
+  public void tearDown() {
+    ApplicationManager.setApplication(application, disposable);
   }
 
   @Test
@@ -82,7 +83,7 @@ public class CloudToolsPluginInitializationComponentTest extends BasePluginTestC
   public void testInitComponent_usageTrackingIsEnabled() {
     Application mockApplication = spy(ApplicationManager.getApplication());
     when(mockApplication.isUnitTestMode()).thenReturn(false);
-    ApplicationManager.setApplication(mockApplication, mock(Disposable.class));
+    ApplicationManager.setApplication(mockApplication, disposable);
     testComponent = spy(new CloudToolsPluginInitializationComponent());
     doNothing().when(testComponent).configureUsageTracking();
     testComponent.initComponent();
@@ -93,7 +94,7 @@ public class CloudToolsPluginInitializationComponentTest extends BasePluginTestC
   public void testInitComponent_usageTrackingIsDisabled() {
     Application mockApplication = spy(ApplicationManager.getApplication());
     when(mockApplication.isUnitTestMode()).thenReturn(true);
-    ApplicationManager.setApplication(mockApplication, mock(Disposable.class));
+    ApplicationManager.setApplication(mockApplication, disposable);
     testComponent = spy(new CloudToolsPluginInitializationComponent());
     testComponent.initComponent();
     verify(testComponent, never()).configureUsageTracking();

